petab_select/candidate_space.py

```diff
@@ -722,7 +708,7 @@ def is_plausible(self, model: Model) -> bool:
         # A model is plausible if the number of estimated parameters strictly
         # increases (or decreases, if `self.direction == -1`), and no
         # previously estimated parameters become fixed.
-        if self.predecessor_model == VIRTUAL_INITIAL_MODEL or (
+        if self.predecessor_model.hash == VIRTUAL_INITIAL_MODEL.hash or (
```

Comment:
I thought that explicitly checking hashes would be better, because then it's clear that the equality is only up to the information in the hash (model subspace ID and indices). An equality check between models could mean multiple things:
- the position in the model space is equal (same hashes)
- the models contain all the same information (position in model space / estimated parameters / criteria)
- the models are equal in some / all criteria, for comparison during model selection (i.e. then I could also implement `<`, `>`)

So to avoid confusion to users I decided to not implement `__eq__`. I could raise a `NotImplementedError` and point users to the other comparison methods (`Model.hash` or `petab_select.model.default_compare`).

re: model hashes
Since `__hash__` truncates the hash string, I disabled `Model.__hash__` and point users to `Model.hash` instead.
I implemented `ModelHash.__hash__` and `ModelHash.__eq__` just so it can be used as keys in a dictionary.